new-packages/Core/src/String.candy

```diff
@@ -20,11 +20,21 @@ impl String {
   public builtin fun trim(): String
 }
 
-builtin impl String: Equals & Comparable
+impl String: Equals {
+  ## TODO(marcelgarus): Implement `Compare` instead, which automatically implements `Equals`.
+
+  fun equals(other: String): Bool {
+    this.length() == other.length()
+      & this.characters().zip(other.characters()).all({ it.first == it.second })
```

Comment:
Unfortunately, that won't work – the items returned by `characters()` are `String`s themselves, leading to infinite recursion. Until we handle Unicode in Candy directly, we'll probably have to make this `builtin`.